crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -590,6 +590,18 @@ impl<'w> UnsafeWorldCell<'w> {
             .get(component_id)?
             .get_with_ticks()
     }
+
+    // Returns a mutable reference to the underlying world's [`CommandQueue`].
+    /// # Safety
+    /// It is the callers responsibility to ensure that
+    /// - the [`UnsafeWorldCell`] has permission to access the queue mutably
+    /// - no mutable references to the queue exist at the same time
+    pub(crate) unsafe fn get_command_queue(self) -> &'w mut CommandQueue {
+        // SAFETY:
+        // - caller ensures there are no existing mutable references
+        // - caller ensures that we have permission to access the queue
+        unsafe { &mut *addr_of_mut!((*self.0).command_queue) }
```

Comment:
Not super comfortable with this, this the rest of `UnsafeWorldCell`'s method don't use or respect this memory access pattern (`world_mut` is a special case). They just create and encapsulate `&World` references and provide restricted access. It's fine for now though since it's private and unsafe, but I think we should refactor the rest of `UnsafeWorldCell` to use this memory access pattern, and probably remove `world_metadata` since it's possible to access metadata safely through provided methods.